worker/uniter/uniter.go

```diff
@@ -236,6 +236,9 @@ func (u *Uniter) init(unitTag names.UnitTag) (err error) {
 	u.storage = storageAttachments
 	u.addCleanup(storageAttachments.Stop)
 
+	if err := charm.ClearDownloads(u.paths.State.BundlesDir); err != nil {
+		return errors.Annotatef(err, "cannot clear downloads")
```

Comment:
I'd prefer if we would just bail, but let's warn for now to be on the safe side. Something external could block the deletion of the directory/files by holding onto them. Shouldn't happen, but it's possible, and I don't think we should risk that in a maintenance branch.